src/monkey_patch/trackers/buffered_logger.py

```diff
@@ -14,6 +15,12 @@
 LIB_NAME = "monkey-patch"
 ENVVAR = "MONKEY_PATCH_LOG_DIR"
 
+# -- Dataset Length --  
+
+class Dataset(BaseModel):
+    alignments: str
```

Comment:
Can you please revisit this? The correct structure should look something like:
```
class Dataset(BaseModel):
    alignments: Dict[str, int] = {}
    patches: Dict[str, int] = {}
```

As you can see, we need to get the dataset for each registered function - as these are all tracked differently.

Additionally, this object is referenced extensively in `function_modeler.py` - which will need to be refactored.

For instance, in line 67 in `function_modeler`, we have:
```
self.dataset_sizes["alignments"][function_hash] = 1
```

Which in this new implementation would look like:
```
self.dataset_sizes.alignments[function_hash] = 1
```